src/NuGet.Core/NuGet.Protocol/RemoteRepositories/HttpFileSystemBasedFindPackageByIdResource.cs

```diff
@@ -161,24 +198,37 @@ public class HttpFileSystemBasedFindPackageByIdResource : FindPackageByIdResourc
                 throw new ArgumentNullException(nameof(logger));
             }
 
-            cancellationToken.ThrowIfCancellationRequested();
+            var stopwatch = Stopwatch.StartNew();
+            try
+            {
+                cancellationToken.ThrowIfCancellationRequested();
 
-            var packageInfos = await EnsurePackagesAsync(id, cacheContext, logger, cancellationToken);
+                var packageInfos = await EnsurePackagesAsync(id, cacheContext, logger, cancellationToken);
 
-            PackageInfo packageInfo;
-            if (packageInfos.TryGetValue(version, out packageInfo))
-            {
-                var reader = await _nupkgDownloader.GetNuspecReaderFromNupkgAsync(
-                    packageInfo.Identity,
-                    packageInfo.ContentUri,
-                    cacheContext,
-                    logger,
-                    cancellationToken);
+                PackageInfo packageInfo;
+                if (packageInfos.TryGetValue(version, out packageInfo))
+                {
+                    var reader = await _nupkgDownloader.GetNuspecReaderFromNupkgAsync(
+                        packageInfo.Identity,
+                        packageInfo.ContentUri,
+                        cacheContext,
+                        logger,
+                        cancellationToken);
 
-                return GetDependencyInfo(reader);
-            }
+                    return GetDependencyInfo(reader);
+                }
 
-            return null;
+                return null;
+            }
+            finally
```

Comment:
This might be tricky to reason about.
In this call is where pretty much all the downloads from V3 feeds happen, because we call the nupkg downloader.

Reply:
Talked offline. The reason that all the `FindPackageById` methods are instrumented in all implementations is to avoid these types of issues where `GetDependencyInfo` downloads a nupkg in one implementation, but not in another. Only the sum of all resource event times will be comparable, even if `GetDependencyInfo` events are not.